RecoTracker/TkSeedGenerator/plugins/SeedProducerFromCuda.cc

```diff
@@ -0,0 +1,196 @@
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/TrajectoryState/interface/LocalTrajectoryParameters.h"
+#include "DataFormats/GeometrySurface/interface/Plane.h"
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+#include "DataFormats/TrackingRecHit/interface/InvalidTrackingRecHit.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+#include "HeterogeneousCore/CUDACore/interface/GPUCuda.h"
+#include "HeterogeneousCore/CUDAServices/interface/CUDAService.h"
+#include "HeterogeneousCore/Producer/interface/HeterogeneousEDProducer.h"
+#include "RecoPixelVertexing/PixelTriplets/plugins/pixelTuplesHeterogeneousProduct.h"
+
+#include "Geometry/CommonDetUnit/interface/GeomDet.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "TrackingTools/MaterialEffects/interface/PropagatorWithMaterial.h"
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+#include "TrackingTools/AnalyticalJacobians/interface/JacobianLocalToCurvilinear.h"
+#include "TrackingTools/TrajectoryParametrization/interface/GlobalTrajectoryParameters.h"
+#include "TrackingTools/TrajectoryParametrization/interface/CurvilinearTrajectoryError.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+
+#include "RecoPixelVertexing/PixelTrackFitting/interface/FitUtils.h"
+
+/*
+  produces seeds directly from cuda produced tuples
+*/
+class SeedProducerFromCuda
```

Comment:
In principle it would be more clear to call this class to e.g. "SeedProducerFromSOA" since is not transferring data from GPU, but using a SOA that already is on the CPU (transferred by CAHitQuadrupletGeneratorGPU). Same holds actually for PixelTrackProducerFromCUDA as well.

On the other hand, all of these need to be reorganized anyway (hopefully sooner than later), so I'm ok to continue with this (somewhat misleading to me) name.

Reply:
definitively!
will change either now or when moving to the real SOA for pixeltracks